### PR TITLE
fix: omit session duration parameter when a duration of zero is used

### PR DESCRIPTION
**Pull request by joshdk**

### CODE REVIEW

**Review by jimethn — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -116,6 +116,13 @@ Attach a readonly policy to the federated user:
 aws-console --policy readonly
 ```
 
+### Caveats
+
+- If the profile used refers to a role that was assumed using user session credentials (obtained via `aws sts get-session-token`) then you **must** specify a session duration of zero seconds.
```
Comment (jimethn) [inline README.md:121]: I'd prefer `duration` to default to 0 if unspecified. If you pass no duration, then the console session lasts as long as the ephemeral credentials used. Probably that's what most people want anyway.

Comment (joshdk) [reply README.md:121]: The Console session "API" is actually pretty unique in that you can start with very short lived credentials (say, that are valid for an hour) and trade them for a much longer Console session (say 12 hours).

I would say that the desired use-case is that someone generates the shortest lived credentials possible, then immediately generates a Console session that they can use for their workday (hence the current 12h default duration). They also then don't need to constantly access their long lived credentials to regenerate new short lived credentials.

Comment (jimethn) [reply README.md:121]: Ah, I didn't realize you could create a console session that lasts longer than the credentials themselves. Neat.

I see the merit of the use-case you propose, but IMO a default should never result in an error. With this default, 1 of 3 possible credential methods (permanent, session, federated) will fail. Either the default should be something that works with all 3 methods, or the program should detect if the user is using an incompatible method and automatically adjust the API call.

Just my $0.02 :)

Comment (joshdk) [reply README.md]: Circling back to this after ~4 months - I switched `--duration` to have a default of 0, and verified that all the various credential flows work out of the box without requiring anything custom.

**Pull request merged**